app/models/endpoint.rb

```diff
@@ -11,4 +11,24 @@ class Endpoint < ApplicationRecord
   validates :endpoint_node, presence: true
   validates :storage_location, presence: true
   validates :recovery_cost, presence: true
+
+  # rubocop:disable Metrics/AbcSize
```

Comment:
this inline disables aren't my favorite, but it's certainly not worth more effort to simply shut up rubocop.  Shameless green is fine.

However, I did find it a bit more work than I would have liked to puzzle out the method below, and I'm wondering if my suggestions will help; and if they do, will they appease rubocop.